### DISCUSSION TITLE
Showing entire process tree in search

### DISCUSSION THREAD

**Discussion by sid-6581** [winsiderss/systeminformer#2252] [Questions / Answers]
One of the things I need to do fairly often is finding child processes that are either left behind after the parent process has exited, or parent processes that are stuck trying to detect the exit of a child process that's already gone. It would be great to filter processes and show the entire process tree where either the parent process or child process are present. Is that possible currently?

**Comment by Advait251206** [comment]
I don't believe System Informer's search/filter feature currently works that way.

As far as I've seen, the process search filters the displayed processes based on the matching criteria, but it does not automatically expand the results to include the full parent/child hierarchy of matching processes.

For your use case, the ideal behavior would be something like:

- Search for a process name, PID, command line, etc.
- If a process matches, display its ancestors up to the root.
- Optionally display all descendants as well.
- Visually highlight the matching process within the tree.

That would make it much easier to investigate:

- Orphaned child processes.
- Parent processes waiting on children that have exited.
- Complex process trees (browsers, IDEs, build systems, containers, etc.).
- Process lifetime and ownership issues.

At the moment, I think the closest workaround is to:

1. Find the process using search.
2. Clear the search/filter.
3. Navigate to the process in the normal tree view and manually inspect its parent/children.

A useful enhancement could be a search mode such as:

- **Show matching processes only** (current behavior)
- **Show matching processes and ancestors**
- **Show matching processes and full tree**

I don't recall seeing an existing option for this, but it would be a valuable addition for troubleshooting process hierarchy issues.
